### PR TITLE
Initial code for free hand drawing feature

### PR DESCRIPTION
**Pull request by Shapokhan**
#37

### CODE REVIEW

File: src/Feedback.js [hunk 1]
```diff
@@ -203,25 +213,42 @@ export default function Feedback(argument) {
...
+            imageParentRef.current.offsetLeft,
+          startY:
+            event.clientY +
+            imageScrollOwnerRef.current.scrollTop -
+            imageParentRef.current.offsetTop,
+        };
+      }
+      if (selectedPointer === "Drawing") {
+        setIsDrawing(true);
+        let { clientX, clientY } = event;
```
Comment (omeryousaf) [inline src/Feedback.js:241]: 👍 destructuring object.

File: src/Feedback.js [hunk 2]
```diff
@@ -396,6 +454,7 @@ export default function Feedback(argument) {
                 // onClick={e => selectedPointer !== "Line" ? addNewElement(e) : e.stopPropagation()}
                 onMouseDown={handleEvent}
```
Comment (omeryousaf) [inline src/Feedback.js:455]: @Shapokhan i think we should have separate functions to handle MouseDown and MouseUp events, but not necessary in this PR.

File: src/Feedback.js [hunk 3]
```diff
@@ -404,6 +463,10 @@ export default function Feedback(argument) {
                   className={`${classes.restrictDimensions} ${classes.displayBlock}`}
                 />
                 <svg className="overlay" width="100%" height="100%">
+                  {/* To show the line while drawing */}
+                  {isDrawing && <Drawing drawingPath={drawingPath} />}
```
Comment (omeryousaf) [inline src/Feedback.js:467]: @Shapokhan what if i try to drag another element on the page while the `Draw` option is selected ?

https://github.com/omeryousaf/image-editor-for-ui-feedback/assets/6904200/1108ccc6-a222-421d-b955-f3bfe0d693e8

Comment (Shapokhan) [reply src/Feedback.js:467]: @omeryousaf Yes, there is some issue. Let me work on it.

Comment (omeryousaf) [reply src/Feedback.js]: Fixed. 🎉

File: src/components/Drawing/Drawing.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import { makeStyles } from "@material-ui/core/styles";
+
+const useStyles = makeStyles((theme) => ({
+  lineCursor: { cursor: "all-scroll" },
+}));
+
+const Drawing = (props) => {
```
Comment (omeryousaf) [inline src/components/Drawing/Drawing.js:8]: @Shapokhan `Drawing` sounds a bit vague, let's rename the component and file to `FreeHandDrawing`. What do u think ?

**Comment by Shapokhan** [comment]
@omeryousaf the issue of not adding a pin to an image is now fixed. 
But when I clicked on the existing pin, a new pin is adding instead of making the existing one selected. Should I work on it in this branch or we should log a new issue for this? As the application is now working the same as the live application.

**Comment by Shapokhan** [comment]
> @omeryousaf the issue of not adding a pin to an image is now fixed. But when I clicked on the existing pin, a new pin is adding instead of making the existing one selected. Should I work on it in this branch or we should log a new issue for this? As the application is now working the same as the live application.

I have reopened issue-4 and create new pull request to fix this problem of "Click on existing pin will make it selected instead of new pin on top of it".

**Review by omeryousaf — APPROVED** [review]

**Pull request merged**